src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -77,6 +62,12 @@ public void OnHttpRequestInStart(HttpContext httpContext)
         {
             if (this.client.IsEnabled())
             {
+                if (Activity.Current == null)
+                {
+                    AspNetCoreEventSource.Instance.LogHostingDiagnosticListenerOnHttpRequestInStartActivityNull();
+                    return;
+                }
+
                 var requestTelemetry = InitializeRequestTelemetryOnHttpRequestInStart(httpContext);
```

Comment:
I still believe  InitializeRequestTelemetryOnHttpRequestInStart and InitializeRequestTelemetryOnBeginRequest are almost the same:
Basically InitializeRequestTelemetryOnBeginRequest  is 
```
InitializeRequestTelemetryOnBeginRequest (context, timestamp)
{
  var a = new Activity(...);
  // get Request-Id and Correlation-Context headers
  a.Start();
  InitializeRequestTelemetryOnHttpRequestInStart (context, timestamp)
}
```